### PR TITLE
chore: Refactor string span management

### PR DESCRIPTION
**Pull request by dranikpg**
Pulls iterator support from helio into the dragonfly repo. Also removes big chunk of #3154

### CODE REVIEW

File: src/facade/facade_types.h [hunk 1]
```diff
@@ -46,6 +49,44 @@ inline std::string_view ToSV(std::string_view slice) {
   return slice;
 }
 
+inline std::string_view ToSV(const std::string& slice) {
```
Comment (BorysTheDev) [inline src/facade/facade_types.h:52]: Could you prevent to use it with rvalue

File: src/facade/facade_types.h [hunk 2]
```diff
@@ -46,6 +49,44 @@ inline std::string_view ToSV(std::string_view slice) {
...
+inline std::string_view ArgS(CmdArgList args, size_t i) {
+  auto arg = args[i];
+  return {arg.data(), arg.size()};
+}
+
+constexpr auto kToSV = [](auto v) { return ToSV(v); };
+
+struct ArgRange {
+  template <typename T,
+            typename = typename std::enable_if_t<!std::is_base_of_v<ArgRange, std::decay_t<T>>>>
```
Comment (BorysTheDev) [inline src/facade/facade_types.h:65]: I think a  better approach define move and copy constructors separately and you don't need this enable_if in this case

File: src/facade/facade_types.h (hunk 1 above)
Comment (dranikpg) [reply src/facade/facade_types.h:52]: Added a delete

File: src/facade/facade_types.h (hunk 2 above)
Comment (dranikpg) [reply src/facade/facade_types.h]: Yep, seems simpler

File: src/facade/facade_types.h [hunk 3]
```diff
@@ -46,6 +49,50 @@ inline std::string_view ToSV(std::string_view slice) {
   return slice;
 }
 
+inline std::string_view ToSV(const std::string& slice) {
+  return slice;
+}
+
+inline std::string_view ToSV(std::string&& slice) = delete;
+
+constexpr auto kToSV = [](auto&& v) { return ToSV(v); };
```
Comment (kostasrim) [inline src/facade/facade_types.h:58]:
```suggestion
inline std::string_view ToSV(const std::string& slice) {
  return slice;
}

inline std::string_view ToSV(std::string&& slice) = delete;

constexpr auto kToSV = [](auto&& v) { return ToSV(std::forward<decltype(v)>(v)); };
```

Comment (kostasrim) [reply src/facade/facade_types.h:58]: Th expression `ToSV(v)` treats `v` as an lvalue

Comment (dranikpg) [reply src/facade/facade_types.h:58]: True, missed that one 🤓

**Review by kostasrim — APPROVED** [review]

**Pull request merged**